Pages/Index.cshtml.cs

```diff
@@ -114,15 +113,15 @@ private async Task<List<string>> FetchDataFromUrl(string url)
                         return dataList;
                     }
                     else
-                    {
-
+                    { 
                         return new List<string>();
                     }
                 }
             }
-            catch (Exception) // Resolved warnings
+            catch (Exception ex) // Resolved warnings
             {
-                return new List<string>();
+                _logger.LogError("Error: ", ex);
+                return new List<string>{"No Results Found"};
```

Comment:
Logging in an exception block is a good idea.
Stuffing "No Results Found" into JSON... hmmmm, interesting idea, though I'd consider a few alternatives:
- Return an empty collection, and let the UI handle the message.  This is a good idea, as the UI layer can handle internationalization; not this layer.
- Throw an exception, with that message.  That will inform the UI layer, and then, again, the UI layer can decide how best to inform the user.